sdk/src/wallet/operations/block.rs

```diff
@@ -43,4 +40,34 @@ where
 
         Ok(block_id)
     }
+
+    /// Attempt to submit a block, but only if the current congestion allows it without
+    /// overspending block issuance credits.
+    pub(crate) async fn try_submit_basic_block(
+        &self,
+        payload: impl Into<Option<Payload>> + Send,
+        issuer_id: impl Into<Option<AccountId>> + Send,
+    ) -> Result<BlockId> {
+        // If an issuer ID is provided, use it; otherwise, use the first available account or implicit account.
+        let issuer_id = match issuer_id.into() {
+            Some(id) => id,
+            None => self.data().await.first_account_id().ok_or(Error::AccountNotFound)?,
+        };
+
+        let payload = payload.into();
+        let protocol_parameters = self.client().get_protocol_parameters().await?;
+        let mut work_score = protocol_parameters.work_score_parameters().block();
```

Comment:
This doesn't feel very future proof tbh. A solution could be to not have `try_submit_basic_block` but add a flag to `submit_basic_block` to check congestion right before `post_block` so we actually have the block. Do whatever you want with this information, I don't feel very strongly. And before you say, I do think that the cost of building the block is negligible.